ttnn/cpp/ttnn/operations/eltwise/ternary/common/ternary_op_types.hpp

```diff
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <variant>
+
+namespace ttnn::operations::ternary {
+
+// Type alias for scalar or tensor values
+// Supports float, int, and Tensor types
+using TensorScalarVariant = std::variant<float, int, Tensor>;
+
+// Ternary operation types
+enum class TernaryOpType {
+    WHERE,  // conditional selection: out = predicate ? value_true : value_false
+    LERP,   // linear interpolation: out = input + weight * (end - input)
+};
+
+// Variant types for ternary operations
+enum class TernaryVariant {
+    TTT,  // tensor-tensor-tensor
+    TTS,  // tensor-tensor-scalar
+    TST,  // tensor-scalar-tensor
+    TSS,  // tensor-scalar-scalar
+};
```

Comment:
While uninteresting for the specific case of `WHERE`, `STT`, `STS`, and `SST` would be useful to have for operations like `LERP`, which is another reason why I'm suggesting we should consider delegating these cases to the other primitive operations instead.